### PR TITLE
Add Animator.AnimatorPauseListener support

### LINKED ISSUE #1644: addAnimatorPauseListener() support
*Lottie is supported and developed on nights and weekends. Issues from [Lottie sponsors](https://github.com/users/gpeal/sponsorship) will be prioritized.*

**Is your feature request related to a problem? Please describe.**
Using lottie for Android 3.4.4, I'm trying to play a sound in sync with an animation. But I didn't find a reliable way to detect when the animation pause/resume, as the `LottieAnimationView` does not provide callback on these actions.

**Describe the solution you'd like**
Android provide an `Animator.AnimatorPauseListener` interface alongside `Animator.AnimatorListener`.

`LottieAnimationView` API already supports `Animator.AnimatorListener`, but it does not supports `Animator.AnimatorPauseListener`

Supporting this kind of listener would be a natural extension of the `LottieAnimationView` API.

`LottieAnimationView.addAnimatorPauseListener()`
`LottieAnimationView.removeAnimatorPauseListener()`

**Describe alternatives you've considered**
Actually, I'm subclassing LottieAnimationView to generate and dispatch pause/resume events when pauseAnimation()/resumeAnimation() is invoked, but it does not allow me to reuse the `Animator.AnimatorPauseListener` interface as I don't have access to the `ValueAnimator` that should be passed to the callback, and it would be better if implemented directly on the underlying drawable.

### PR DESCRIPTION
**Pull request by heitorcolangelo**
Resolves #1644 by adding methods to LottieAnimationView and LottieDrawable

### CODE REVIEW

File: lottie/src/main/java/com/airbnb/lottie/LottieDrawable.java [hunk 1]
```diff
@@ -742,6 +743,22 @@ public void removeAllAnimatorListeners() {
     animator.removeAllListeners();
   }
 
+  public void addAnimatorPauseListener(Animator.AnimatorPauseListener listener) {
```
Comment (gpeal) [inline lottie/src/main/java/com/airbnb/lottie/LottieDrawable.java:746]: I think it's safer to add `@RequiresApi(api = Build.VERSION_CODES.KITKAT)` to the public API here. Most people don't pay attention to the logs.

**Review by gpeal — COMMENTED** [review]
You'll also have to add the same corresponding methods in `LottieAnimationView` that just call through to `LottieDrawable` You can follow the pattern that lots of other methods there do.

File: lottie/src/main/java/com/airbnb/lottie/LottieDrawable.java (hunk 1 above)
Comment (heitorcolangelo) [reply lottie/src/main/java/com/airbnb/lottie/LottieDrawable.java:746]: I agree, should I remove the `if`statement then?

**Comment by heitorcolangelo** [comment]
@gpeal I don't think I got what you mean by "add the same corresponding methods in `LottieAnimationView`", I did add two methods to `LottieAnimationView`. Can you please be more specific about what I'm missing?

File: lottie/src/main/java/com/airbnb/lottie/LottieDrawable.java (hunk 1 above)
Comment (gpeal) [reply lottie/src/main/java/com/airbnb/lottie/LottieDrawable.java:746]: @heitorcolangelo Oh you're right. I hadn't had my coffee yet!

File: lottie/src/main/java/com/airbnb/lottie/LottieDrawable.java [hunk 2]
```diff
@@ -742,6 +743,22 @@ public void removeAllAnimatorListeners() {
...
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
```
Comment (gpeal) [inline lottie/src/main/java/com/airbnb/lottie/LottieDrawable.java:747]: @heitorcolangelo We should add the same `@RequiresApi` here too. This is also a public API.

Comment (heitorcolangelo) [reply lottie/src/main/java/com/airbnb/lottie/LottieDrawable.java:747]: @gpeal
 And do you think we can remove this if statement? I believe it is not necessary as we added the annotation.
The final version would be just
```kotlin
@RequiresApi(api = Build.VERSION_CODES.KITKAT)
public void addAnimatorPauseListener(Animator.AnimatorPauseListener listener) {
  animator.addPauseListener(listener);
}

@RequiresApi(api = Build.VERSION_CODES.KITKAT)
public void removeAnimatorPauseListener(Animator.AnimatorPauseListener listener) {
  animator.removePauseListener(listener);
}
```

**Review by gpeal — APPROVED** [review]

**Pull request merged**